### DISCUSSION TITLE
AWS Graviton (ARM) processor information

### DISCUSSION THREAD

**Discussion by rmuzzini** [webmin/webmin#2248] [Ideas]
hi all.
on AWS Graviton CPUs (based on ARM), cpuinfo does not return the CPU model. Therefore, processor information is not displayed in the webmin dashboard, and the number of available cores is also omitted. But you can get this information using the lscpu command.
I'm not a PERL programmer and the following is just a draft code (I apologize in advance, blushing…), but maybe my example could be inspiring...?
in /usr/share/webmin/proc/linux-lib.pl, i modified line 347
`
return ( $load[0], $load[1], $load[2] );
`
with
`
my @cpu_info = `lscpu 2>&1`;
chomp(@cpu_info);
my %cpu_info_hash;
foreach my $line (@cpu_info) {
   if ($line =~ /^([^:]+):\s*(.*)$/) {
       my $key = $1;
       my $value = $2;
       $key =~ s/^\s+|\s+$//g;
       $cpu_info_hash{$key} = $value;
   }
}
return ( $load[0], $load[1], $load[2], 0, $cpu_info_hash{'Model name'}, $cpu_info_hash{'Vendor ID'}, 0, $cpu_info_hash{'CPU(s)'} );
`

regards. (and thanks for this excellent software that I have been using since 2003…).
re.mu.

**Comment by iliaross** [comment]
Hello,

What is the output of `sensors` command on your system? Do you have it installed?

**Reply by rmuzzini** [reply]
hi.
thank you for your reply.
there is no such command installed by default (ubuntu server 22.04 on aws graviton). after installing it the output is
nvme-pci-0020
Adapter: PCI adapter
Composite: -273.1°C (low = +69.8°C, high = +69.8°C)

**Reply by rmuzzini** [reply]
other info, just in case: https://github.com/amazonlinux/amazon-linux-2023/issues/455
